internal/util/util.go

```diff
@@ -491,3 +491,56 @@ func VSBHasVSBackupName(backup *velerov1api.Backup, snapCont *snapshotv1api.Volu
 
 	return true
 }
+
+func GetVSBWithConditions(volumeSnapshotbackupNS string, volumeSnapshotName string, log logrus.FieldLogger) (bool, error) {
```

Comment:
Do we really need to poll here ? Should we not directly fetch VSB ?

Reply:
This seems like code duplication, lets modify the GetVolumeSnapshotBackupWithStatusData function itself,
like once you get vsb with status data, check for condition using `checkVSBFailed` func and fail the backup there itself.